src/python/gudhi/representations/vector_methods.py

```diff
@@ -379,72 +383,88 @@ def transform(self, X):
         if not self.is_fitted():
             raise NotFittedError("Not fitted.")
 
-        if not X:
-            X = [np.zeros((0, 2))]
-        
         N = len(X)
 
-        events = np.concatenate([pd.ravel(order="F") for pd in X], axis=0)
-        sorting = np.argsort(events)
-        offsets = np.zeros(1 + N, dtype=int)
-        for i in range(0, N):
-            offsets[i+1] = offsets[i] + 2*X[i].shape[0]
-        starts = offsets[0:N]
-        ends = offsets[1:N + 1] - 1
+        if sum([len(x) for x in X]) == 0:
+            print("Empty list or empty diagrams: output contains only zeros")
+            if self.resolution is None:
+                if not self.is_fitted():
+                    self.grid_ = np.array([-np.inf])
+                return np.zeros((N, 1))
+            else:
+                if not self.is_fitted():
+                    self.grid_ = np.full(shape=[self.resolution], fill_value=-np.inf)
+                return np.zeros((N, self.resolution))
```

Comment:
```suggestion
        if N == 0:
            return np.zeros((N, len(self.grid_)))
```

Reply:
I see, that makes sense. Just pushed a fix.